MoreLinq.Test/TagFirstLastTest.cs

```diff
@@ -22,12 +22,29 @@ namespace MoreLinq.Test
     [TestFixture]
     public class TagFirstLastTest
     {
+        [Test]
+        public void TagFirstLastEvaluatesSourceLazily()
+        {
+            var source = MoreEnumerable.From(() => 123, () => 456, () => throw new TestException());
+            source.TagFirstLast((item, isFirst, isLast) => new { Item = item, IsFirst = isFirst, IsLast = isLast })
+                  .Take(1)
+                  .Consume();
+        }
+
         [Test]
         public void TagFirstLastIsLazy()
         {
             new BreakingSequence<object>().TagFirstLast(BreakingFunc.Of<object, bool, bool, object>());
         }
 
+        [Test]
+        public void TagFirstLastWithSourceSequenceOfZero()
+        {
+            var source = Enumerable.Empty<int>();
+            var sut = source.TagFirstLast((item, isFirst, isLast) => new { Item = item, IsFirst = isFirst, IsLast = isLast });
```

Comment:
The lambda will never be called so we could make it more strict and clear like so:

```suggestion
            var sut = source.TagFirstLast(BreakingFunc.Of<int, bool, bool, int>());
```